casper/src/main/scala/coop/rchain/casper/util/comm/ProposeService.scala

```diff
@@ -0,0 +1,50 @@
+package coop.rchain.casper.util.comm
+
+import java.io.Closeable
+import java.util.concurrent.TimeUnit
+
+import scala.util.Either
+
+import coop.rchain.casper.protocol._
+import coop.rchain.models.either.implicits._
+
+import com.google.protobuf.empty.Empty
+import io.grpc.{ManagedChannel, ManagedChannelBuilder}
+import monix.eval.Task
+
+trait ProposeService[F[_]] {
+  def propose(): F[Either[Seq[String], String]]
+}
+
+object ProposeService {
+  def apply[F[_]](implicit ev: ProposeService[F]): ProposeService[F] = ev
+}
+
+class GrpcProposeService(host: String, port: Int, maxMessageSize: Int)
+    extends ProposeService[Task]
+    with Closeable {
+
+  private val channel: ManagedChannel =
+    ManagedChannelBuilder
+      .forAddress(host, port)
+      .maxInboundMessageSize(maxMessageSize)
+      .usePlaintext()
+      .build
+
+  private val stub = ProposeServiceGrpcMonix.stub(channel)
+
+  def propose(): Task[Either[Seq[String], String]] =
+    stub.propose(Empty()).map(_.toEither[DeployServiceResponse].map(_.message))
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  override def close(): Unit = {
+    val terminated = channel.shutdown().awaitTermination(10, TimeUnit.SECONDS)
+    if (!terminated) {
+      println(
```

Comment:
curious as to why not use log.warn?

Reply:
because it is effectful and we don't control that part (api that sucks)